types/picomatch/lib/picomatch-options.d.ts

```diff
@@ -0,0 +1,175 @@
+     */
+    nobrace?: boolean;
+    /**
+     * Disable brace matching, so that `{a,b}` and `{1..3}` would be treated as literal characters.
+     */
+    nobracket?: boolean;
+    /**
+     * Make matching case-insensitive. Equivalent to the regex `i` flag. Note that this option is overridden by the `flags` option.
+     */
+    nocase?: boolean;
+    /**
+     * @deprecated use `nounique` instead.
+     * This option will be removed in a future major release. By default duplicates are removed.
+     * Disable uniquification by setting this option to false.
+     */
+    nodupes?: boolean;
+    /**
+     * Alias for `noextglob`
+     */
+    noext?: boolean;
+    /**
+     * Disable support for matching with extglobs (like `+(a\|b)`)
+     */
+    noextglob?: boolean;
+    /**
+     * Disable support for matching nested directories with globstars (`**`)
+     */
+    noglobstar?: boolean;
+    /**
+     * Disable support for negating with leading `!`
+     */
+    nonegate?: boolean;
+    /**
+     * Disable support for regex quantifiers (like `a{1,2}`) and treat them as brace patterns to be expanded.
+     */
+    noquantifiers?: boolean;
+    /**
+     * Function to be called on ignored items.
+     */
+    onIgnore?: (result: Result) => void;
+    /**
+     * Function to be called on matched items.
+     */
+    onMatch?: (result: Result) => void;
+    /**
+     * Function to be called on all items, regardless of whether or not they are matched or ignored.
+     */
+    onResult?: (result: Result) => void;
+    /**
+     * Support POSIX character classes ("posix brackets").
+     */
+    posix?: boolean;
+    /**
+     * Convert all slashes in file paths to forward slashes. This does not convert slashes in the glob pattern itself
+     */
+    posixSlashes?: boolean;
+    /**
+     * Convert all slashes in file paths to forward slashes. This does not convert slashes in the glob pattern itself
+     */
+    prepend?: boolean;
+    /**
+     * Use regular expression rules for `+` (instead of matching literal `+`), and for stars that follow closing parentheses or brackets (as in `)*` and `]*`).
+     */
+    regex?: boolean;
+    /**
+     * Throw an error if brackets, braces, or parens are imbalanced.
+     */
+    strictBrackets?: boolean;
+    /**
+     * When true, picomatch won't match trailing slashes with single stars.
+     */
+    strictSlashes?: boolean;
+    /**
+     * Remove backslashes preceding escaped characters in the glob pattern. By default, backslashes are retained.
+     */
+    unescape?: boolean;
+    /**
+     * Alias for `posixSlashes`, for backwards compatibility.
+     */
+    unixify?: boolean;
+}
+
+interface Result {
+    glob: string;
+    state: any;
+    regex: RegExp;
+    posix: boolean;
+    input: string;
+    output: string;
+    match: ReturnType<typeof picomatch.test>['match'];
+    isMatch: ReturnType<typeof picomatch.test>['isMatch'];
+}
+
+// Shut off automatic exporting
+export {};
```

Comment:
Should be removed IMHO

Reply:
If I remove this and nothing else, it makes dtslint unhappy. I might be able to rework this file not to use es6 stuff and then remove this, but I do like that this file doesn't imply it has anything but types in it. There is no corresponding file for this in picomatch source, so people would ideally only `import type` from this so no 'require()'s wind up in the compiled output.